### ISSUE TITLE
tIED: insertIed incorrectly orders elements within the DataTypeTemplates

### ISSUE THREAD

**Issue by danyill** [OpenEnergyTools/scl-lib#47]
Similar to #46, it seems to be possible to incorrectly add DataTypeTemplates and create a schema invalid file.

I've seen for example the following in my DataTypeTemplate section:

```xml
<LNodeType id="SIPROTEC5_LNType_ATCC_Parallel_Goose_Proxy" lnClass="ATCC">
      <DO name="Mod" type="SIPROTEC5_DOType_Mod_controllable_with_InitValVis_for_SIEMENS_LN_V09.50.01_V09.50.01"/>
      <DO name="Blk" type="SIPROTEC5_DOType_SPS_for_Input_V09.50.01_V09.50.01"/>
      <DO name="Auto" type="SIPROTEC5_DOType_SPC_ATCC_Auto_V09.50.01_V09.50.01"/>
      <DO name="ParOp" type="SIPROTEC5_DOType_SPC_statusOnly_and_dir_with_nor_sec_V09.50.01_V09.50.01"/>
      <DO name="TapChg" type="SIPROTEC5_DOType_BSC_controllable_without_BinaryInput_V09.50.01_V09.50.01"/>
      <DO name="CtlV" type="SIPROTEC5_DOType_MV_Float_V09.50.01_V09.50.01"/>
      <DO name="LodA" type="SIPROTEC5_DOType_MV_Float_V09.50.01_V09.50.01"/>
      <DO name="PhAng" type="SIPROTEC5_DOType_MV_Float_V09.50.01_V09.50.01"/>
      <DO name="Beh" type="SIPROTEC5_DOType_ENS_Behavior_V09.50.01_V09.50.01"/>
      <DO name="Health" type="SIPROTEC5_DOType_ENS_Health_V09.50.01_V09.50.01"/>
      <DO name="NamPlt" type="SIPROTEC5_DOType_LPL_NamPlt_for_Ed2_LN_V09.50.01_V09.50.01"/>
      <DO name="RHealth" type="SIPROTEC5_DOType_INS_INT32_with_dataNs_V09.50.01_V09.50.01"/>
      <DO name="ParGrp" type="SIPROTEC5_DOType_INC_controllable_with_dataNs_V09.50.01_V09.50.01"/>
      <DO name="PTrfBeh" type="SIPROTEC5_DOType_INS_INT32_with_dataNs_V09.50.01_V09.50.01"/>
      <DO name="PTrfId" type="SIPROTEC5_DOType_INS_INT32_with_dataNs_V09.50.01_V09.50.01"/>
      <DO name="IncParGrp" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01"/>
      <DO name="PaAutoBlk" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01"/>
      <DO name="ErrTapDif" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01"/>
      <DO name="RxRHealth1" type="SIPROTEC5_DOType_INS_INT32_with_dataNs_V09.50.01_V09.50.01"/>
      <DO name="RxBlk1" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01"/>
      <DO name="RxAuto1" type="SIPROTEC5_DOType_ComProxy_SPC_V09.50.01_V09.50.01"/>
      <DO name="RxParOp1" type="SIPROTEC5_DOType_ComProxy_SPC_V09.50.01_V09.50.01"/>
      <DO name="RxParGrp1" type="SIPROTEC5_DOType_ComProxy_INC_V09.50.01_V09.50.01"/>
      <DO name="RxPTrfBeh1" type="SIPROTEC5_DOType_INS_INT32_with_dataNs_V09.50.01_V09.50.01"/>
      <DO name="RxTapChg1" type="SIPROTEC5_DOType_ComProxy_BSC_V09.50.01_V09.50.01"/>
      <DO name="RxPTrfId1" type="SIPROTEC5_DOType_INS_INT32_with_dataNs_V09.50.01_V09.50.01"/>
      <DO name="RxCtlV1" type="SIPROTEC5_DOType_ComProxy_MV_V09.50.01_V09.50.01"/>
      <DO name="RxInvX1" type="SIPROTEC5_DOType_ComProxy_MV_V09.50.01_V09.50.01"/>
      <DO name="RxLodA1" type="SIPROTEC5_DOType_ComProxy_MV_V09.50.01_V09.50.01"/>
      <DO name="RxPhAng1" type="SIPROTEC5_DOType_ComProxy_MV_V09.50.01_V09.50.01"/>
    </LNodeType><EnumType id="SIPROTEC5_Textlist_POWCap_V09.50.01_V09.50.01">
      <EnumVal ord="1">None</EnumVal>
      <EnumVal ord="2">Close</EnumVal>
      <EnumVal ord="3">Open</EnumVal>
      <EnumVal ord="4">Close and Open</EnumVal>
    </EnumType>
```

Here is an example file I've prepared after opening an ICD file with a 7SL86 device in it and then using the "Import Templates" plugin to add a SEL-487E-5 device.

There are many schema errors which seem to stem from either incorrectly ordered elements or perhaps a few missing elements.

![image](https://github.com/OpenEnergyTools/scl-lib/assets/674783/ecc61000-088a-4729-a525-d9e88757150e)

[Siprotec_7SL86_Base (1) (2).icd.zip](https://github.com/OpenEnergyTools/scl-lib/files/12653888/Siprotec_7SL86_Base.1.2.icd.zip)

I have not figured out exactly what is happening here. If it's not obvious I can look at this in more detail.

**Comment by JakobVogelsang** [comment]
Same here we need to statically save the reference on importing IEDs. Tricky but doable :). Thank you for spotting.

**Comment by JakobVogelsang** [comment]
The file you have uploaded is an ICD file and I assume that this is the one you want to import, right? 

Under this assumtion:
The file has already some schema issue as the order is not correct. Still, the way `importIED` works at the moment, DataTypeTemplates are copied over and re-ordered through that. When I open the file on [scl-editor](https://openenergytools/scl-editor) I find a lot of errors and when I import those there is only one left showing the sequence issue of the Communication element.

**Comment by JakobVogelsang** [comment]
I think this part should work correctly and have added some test https://github.com/OpenEnergyTools/scl-lib/commit/a6577af4b11c242a8ceda1e89e0f5ae4363556a1 here. Can you try and run this test with release 0.5.11 ?

**Comment by danyill** [comment]
Sure, I can try. The file was the result of some import using the insertIed
function. I will retest. Thanks for the super-quick response 👍

**Comment by danyill** [comment]
There is still a problem, I will do a PR :wink:

**Comment by danyill** [comment]
The simplest part of the problem is an incorrect element reference which I have done a PR for.

The harder part, is, if there is two different datatypes such that a new datatype has to be generated, then if that is every checked for an IED connection because the edits haven't been made yet, it will found to not be part of the IED connection.

There is two ICD files attached.

When I use my plugin to rename them and add them both in, I get missing datatypes.

[Siprotec_.zip](https://github.com/OpenEnergyTools/scl-lib/files/12702123/Siprotec_.zip)

For example, this `LNodeType`:

```xml
<LNodeType id="SIPROTEC5_LNType_XCBR_1pol_Ctl" lnClass="XCBR">XX
  <DO name="Mod" type="SIPROTEC5_DOType_Mod_status_only_with_InitVal_V09.50.01_V09.50.01" />
  <DO name="Beh" type="SIPROTEC5_DOType_ENS_Behavior_V09.50.01_V09.50.01" />
  <DO name="Health" type="SIPROTEC5_DOType_ENS_Health_V09.50.01_V09.50.01" />
  <DO name="NamPlt" type="SIPROTEC5_DOType_LPL_NamPlt_for_Ed2_LN_V09.50.01_V09.50.01" />
  <DO name="Loc" type="SIPROTEC5_DOType_SPS_V09.50.01_V09.50.01" />
  <DO name="EEHealth" type="SIPROTEC5_DOType_ENS_Health_V09.50.01_V09.50.01" />
  <DO name="OpCnt" type="SIPROTEC5_DOType_INS_INT32_V09.50.01_V09.50.01" />
  <DO name="Pos" type="SIPROTEC5_DOType_DPC_XCBR_Pos_V09.50.01_V09.50.01" />
  <DO name="BlkOpn" type="SIPROTEC5_DOType_SPC_status_only_V09.50.01_V09.50.01" />
  <DO name="BlkCls" type="SIPROTEC5_DOType_SPC_status_only_V09.50.01_V09.50.01" />
  <DO name="SumSwARs" type="SIPROTEC5_DOType_BCR_noFreeze_V09.50.01_V09.50.01" />
  <DO name="POWCap" type="SIPROTEC5_DOType_ENS_TextList_POWCap_V09.50.01_V09.50.01" />
  <DO name="SetOpCntA" type="SIPROTEC5_DOType_INC_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SetOpCntB" type="SIPROTEC5_DOType_INC_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SetOpCntC" type="SIPROTEC5_DOType_INC_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SetOpnHrs" type="SIPROTEC5_DOType_INC_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SetOpHrs" type="SIPROTEC5_DOType_INC_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SetSwASumA"
    type="SIPROTEC5_DOType_APC_FLOAT32_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SetSwASumB"
    type="SIPROTEC5_DOType_APC_FLOAT32_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SetSwASumC"
    type="SIPROTEC5_DOType_APC_FLOAT32_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="AqBlk" type="SIPROTEC5_DOType_SPS_for_Input_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="Rdy" type="SIPROTEC5_DOType_SPS_for_Input_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="RsSwStat" type="SIPROTEC5_DOType_SPS_for_Input_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="RsAqBlkSub" type="SIPROTEC5_DOType_SPS_for_Input_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="TripRelViaBi" type="SIPROTEC5_DOType_SPS_for_Input_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="TripOpnCmd" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="TripCmdA" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="TripCmdB" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="TripCmdC" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="ClsCmd" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="OpnCmdA" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="OpnCmdB" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="OpnCmdC" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="ClsCmdA" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="ClsCmdB" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="ClsCmdC" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="CmdActv" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="AlmSupp" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="DefTr" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="OpCntPhA" type="SIPROTEC5_DOType_INS_INT32_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="OpCntPhB" type="SIPROTEC5_DOType_INS_INT32_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="OpCntPhC" type="SIPROTEC5_DOType_INS_INT32_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SumSwARsA" type="SIPROTEC5_DOType_BCR_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SumSwARsB" type="SIPROTEC5_DOType_BCR_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SumSwARsC" type="SIPROTEC5_DOType_BCR_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SwAphsA" type="SIPROTEC5_DOType_MV_Float_dupd_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SwAphsB" type="SIPROTEC5_DOType_MV_Float_dupd_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SwAphsC" type="SIPROTEC5_DOType_MV_Float_dupd_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SwANeut" type="SIPROTEC5_DOType_MV_Float_dupd_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SwVphsA" type="SIPROTEC5_DOType_MV_Float_dupd_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SwVphsB" type="SIPROTEC5_DOType_MV_Float_dupd_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SwVphsC" type="SIPROTEC5_DOType_MV_Float_dupd_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="CBOpnHrs" type="SIPROTEC5_DOType_INS_INT32_with_units_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="OpHrs" type="SIPROTEC5_DOType_INS_INT32_with_units_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="BbpBlockDO" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="NoTrpRelBi" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
</LNodeType>
```

Differs from this one:

```xml
<LNodeType id="SIPROTEC5_LNType_XCBR_1pol_Ctl" lnClass="XCBR">YY
  <DO name="Mod" type="SIPROTEC5_DOType_Mod_status_only_with_InitVal_V09.50.01_V09.50.01" />
  <DO name="Beh" type="SIPROTEC5_DOType_ENS_Behavior_V09.50.01_V09.50.01" />
  <DO name="Health" type="SIPROTEC5_DOType_ENS_Health_V09.50.01_V09.50.01" />
  <DO name="NamPlt" type="SIPROTEC5_DOType_LPL_NamPlt_for_Ed2_LN_V09.50.01_V09.50.01" />
  <DO name="Loc" type="SIPROTEC5_DOType_SPS_V09.50.01_V09.50.01" />
  <DO name="EEHealth" type="SIPROTEC5_DOType_ENS_Health_V09.50.01_V09.50.01" />
  <DO name="OpCnt" type="SIPROTEC5_DOType_INS_V09.50.01_V09.50.01" />
  <DO name="Pos" type="SIPROTEC5_DOType_DPC_XCBR_Pos_V09.50.01_V09.50.01" />
  <DO name="BlkOpn" type="SIPROTEC5_DOType_SPC_status_only_V09.50.01_V09.50.01" />
  <DO name="BlkCls" type="SIPROTEC5_DOType_SPC_status_only_V09.50.01_V09.50.01" />
  <DO name="SumSwARs" type="SIPROTEC5_DOType_BCR_noFreeze_V09.50.01_V09.50.01" />
  <DO name="POWCap" type="SIPROTEC5_DOType_ENS_TextList_POWCap_V09.50.01_V09.50.01" />
  <DO name="SetOpCntA" type="SIPROTEC5_DOType_INC_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SetOpCntB" type="SIPROTEC5_DOType_INC_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SetOpCntC" type="SIPROTEC5_DOType_INC_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SetOpnHrs" type="SIPROTEC5_DOType_INC_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SetOpHrs" type="SIPROTEC5_DOType_INC_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SetSwASumA"
    type="SIPROTEC5_DOType_APC_FLOAT32_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SetSwASumB"
    type="SIPROTEC5_DOType_APC_FLOAT32_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SetSwASumC"
    type="SIPROTEC5_DOType_APC_FLOAT32_controllable_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="AqBlk" type="SIPROTEC5_DOType_SPS_for_Input_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="Rdy" type="SIPROTEC5_DOType_SPS_for_Input_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="RsSwStat" type="SIPROTEC5_DOType_SPS_for_Input_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="RsAqBlkSub" type="SIPROTEC5_DOType_SPS_for_Input_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="TripRelViaBi" type="SIPROTEC5_DOType_SPS_for_Input_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="TripOpnCmd" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="TripCmdA" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="TripCmdB" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="TripCmdC" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="ClsCmd" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="OpnCmdA" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="OpnCmdB" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="OpnCmdC" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="ClsCmdA" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="ClsCmdB" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="ClsCmdC" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="CmdActv" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="AlmSupp" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="DefTr" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="OpCntPhA" type="SIPROTEC5_DOType_INS_INT32_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="OpCntPhB" type="SIPROTEC5_DOType_INS_INT32_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="OpCntPhC" type="SIPROTEC5_DOType_INS_INT32_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SumSwARsA" type="SIPROTEC5_DOType_BCR_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SumSwARsB" type="SIPROTEC5_DOType_BCR_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SumSwARsC" type="SIPROTEC5_DOType_BCR_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SwAphsA" type="SIPROTEC5_DOType_MV_Float_dupd_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SwAphsB" type="SIPROTEC5_DOType_MV_Float_dupd_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SwAphsC" type="SIPROTEC5_DOType_MV_Float_dupd_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SwANeut" type="SIPROTEC5_DOType_MV_Float_dupd_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SwVphsA" type="SIPROTEC5_DOType_MV_Float_dupd_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SwVphsB" type="SIPROTEC5_DOType_MV_Float_dupd_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="SwVphsC" type="SIPROTEC5_DOType_MV_Float_dupd_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="CBOpnHrs" type="SIPROTEC5_DOType_INS_INT32_with_units_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="OpHrs" type="SIPROTEC5_DOType_INS_INT32_with_units_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="BbpBlockDO" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="NoTrpRelBi" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
  <DO name="RxPos1" type="SIPROTEC5_DOType_ComProxy_DPC_V09.50.01_V09.50.01" />
  <DO name="RxPosA1" type="SIPROTEC5_DOType_ComProxy_DPC_V09.50.01_V09.50.01" />
  <DO name="RxPosB1" type="SIPROTEC5_DOType_ComProxy_DPC_V09.50.01_V09.50.01" />
  <DO name="RxPosC1" type="SIPROTEC5_DOType_ComProxy_DPC_V09.50.01_V09.50.01" />
  <DO name="WetSup" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
</LNodeType>
```

By:

At least a different `OpCnt` datatype reference

The following additional DOs:

```xml
  <DO name="RxPos1" type="SIPROTEC5_DOType_ComProxy_DPC_V09.50.01_V09.50.01" />
  <DO name="RxPosA1" type="SIPROTEC5_DOType_ComProxy_DPC_V09.50.01_V09.50.01" />
  <DO name="RxPosB1" type="SIPROTEC5_DOType_ComProxy_DPC_V09.50.01_V09.50.01" />
  <DO name="RxPosC1" type="SIPROTEC5_DOType_ComProxy_DPC_V09.50.01_V09.50.01" />
  <DO name="WetSup" type="SIPROTEC5_DOType_SPS_with_dataNs_V09.50.01_V09.50.01" />
```

As a result a new DOType must be created.

But in the combined file, instead the DOType is entirely missing.

In my plugin I am renaming and updating the ConnectedAPs (so the IED is not named template) but then the IED that is being checked in the `insertIED` code is checking for `SIEMENS_7UT85_01SIPROTEC5_LNType_XCBR_1pol_Ctl`

Note that this is the LNodeType `SIPROTEC5_LNType_XCBR_1pol_Ctl` prefixed with the IED name: `SIEMENS_7UT85_01`.

But this doesn't exist because the actions haven't been consumed by the document yet, so the lookup of these within the `isDataTypeConnectionToIed` is false.

Because of this the DOType does not get created.

I hope this makes some sense.

**Comment by danyill** [comment]
Also it would be good at some stage to do `/importIED/insertIED/*` on the paths :wink: 
And `action` => `edit` because we are confusing referring to our somewhat deprecated API but using the its terms in the new one...

**Comment by JakobVogelsang** [comment]
@danyill you are the sh*t! This is why your contribution is so valuable. 

I would love to split your findings a bit up to smaller and simpler units we are talking about and implementing, like so:

1. This issue tracks the initial issue of `DataTypeTemplates` children being out of order. (Fix on the way in #48)
2. The second issue is regarding incorrect setting of references. 
3. Be more precise on naming and avoid `action` to not confuse with the deprecated OpenSCD API.
